storage_service/locations/models/package.py

```diff
@@ -894,6 +896,25 @@ def start_reingest(self, pipeline, reingest_type):
         elif reingest_type == self.METADATA_ONLY:
             reingest_files.append(os.path.join(relative_path, 'data', 'objects', 'metadata', ''))
 
+        # Fetch processing configuration, put it in the root of the package and
+        # include the file in reingest_files.
+        if reingest_type == self.FULL and processing_config != 'default':
+            config = pipeline.get_processing_config(processing_config)
+            if config == '':
+                LOGGER.debug('Reingest: processing configuration %s could not be loaded', processing_config)
+            else:
+                config_path =  os.path.join(local_path, 'processingMCP.xml')
+                try:
+                    # It's not expected to find an existing processingMCP.xml
+                    # file in the original AIP, but we are using the w+ mode
+                    # just in case.
+                    with open(config_path, 'w+') as f:
+                        f.write(config)
+                    LOGGER.debug('Reingest: processing configuration %s written, location: %s', processing_config, config_path)
+                    reingest_files.append(os.path.join(relative_path, 'processingMCP.xml'))
+                except IOError as e:
+                    LOGGER.error('Reingest: processing configuration %s could not be written, error: %s', processing_config, e)
```

Comment:
Tip: `LOGGER.exception` or `LOGGER.error( ... , exc_info=True)` will attach the traceback to the log, instead of just printing the exception-string.